src/Service/Voter.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Entity\Hobby;
+use App\Entity\User;
+use App\Entity\Vote;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\OptimisticLockException;
+use Doctrine\ORM\ORMException;
+
+class Voter
+{
+    /** @var EntityManager */
+    private $em;
+
+    public function __construct(EntityManager $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * @param int $hobbyId
+     * @param string $userEmail
+     * @param int $amount
+     * @return string
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function vote(int $hobbyId, string $userEmail, int $amount): string
+    {
+        /** @var User $user */
+        $user = $this->em->getRepository(User::class)->findOneBy(['email' => $userEmail]);
+
+        if (false === $this->checkAmountAndUpdateUser($user, $amount)) {
+            return 'Insufficient budget';
+        }
+        /** @var Hobby $hobby */
+        $hobby = $this->em->getRepository(Hobby::class)->find($hobbyId);
+
+        if (false === $this->updateHobbyAmount($hobby, $amount)) {
+            return 'Insufficient budget';
+        }
+
+        /** @var Vote $vote */
+        $vote = new Vote();
+
+        $vote->setAmount($amount);
+        $vote->setHobby($hobby);
+        $vote->setUser($user);
+
+        $this->em->persist($vote);
+        $this->em->flush();
+
+        //TODO returns should be modify
+        return 'success';
+    }
+
+    /**
+     * @param User $user
+     * @param int $amount
+     * @return bool
+     * @throws ORMException
```

Comment:
Aš tai dedu @throws tags tik prie to metodo kuris throwina Exceptions:) Kad PHPStorm neverktų, galima nusiimta tą Inspection